### PR TITLE
Add cyclical learning rate schedulers

### PR DESCRIPTION
**Pull request by RaphaelMeudec**
Migrating CyclicLR from keras-team/keras-contrib as discussed in https://github.com/keras-team/keras-contrib/issues/519.

- @gabrieldemarmiesse & @seanpmorgan would love some feedback on implementation
- @seanpmorgan is there any recommendations regarding tests? I plan on taking inspiration from [tf core schedulers](https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/keras/optimizer_v2/learning_rate_schedule_test.py), does it seem ok to you?

### CODE REVIEW

**Comment by RaphaelMeudec** [comment]
@googlebot I signed it!

**Comment by gabrieldemarmiesse** [comment]
I'm surprised. Do you guys have a continuous integration service running for pull requests?

**Comment by RaphaelMeudec** [comment]
> I'm surprised. Do you guys have a continuous integration service running for pull requests?

Seems that CI is triggered by adding the kokoro:force-run tag 

> Maybe some tests would help ensure that no bugs are introduced?

Of course! Wanted to check first that implementation I used seems ok.

**Comment by RaphaelMeudec** [comment]
@seanpmorgan Added the tests, ready for review

**Comment by seanpmorgan** [comment]
Thanks @RaphaelMeudec! Looks like there are some line too long lint errors. Could you run this to auto format the code
`bash tools/run_docker.sh -c 'make code-format`

**Comment by RaphaelMeudec** [comment]
@seanpmorgan This is weird, the `code-format` is not doing any change. Is there a difference between the `code-format` and `sanity-check`? I'll fix the errors tomorrow

**Comment by WindQAQ** [comment]
> @seanpmorgan This is weird, the `code-format` is not doing any change. Is there a difference between the `code-format` and `sanity-check`? I'll fix the errors tomorrow

`yapf` does not aggressively make the length of lines fit 80 characters so you should manually modify it in most of the time.

**Comment by RaphaelMeudec** [comment]
@WindQAQ Fixed the errors, should be good now

**Comment by RaphaelMeudec** [comment]
@seanpmorgan Updated to `register_keras_serializable`, could you rerun kokoro?

File: tensorflow_addons/optimizers/cyclical_learning_rate.py [hunk 1]
```diff
@@ -0,0 +1,305 @@
...
+                      metrics=['accuracy'])
+
+        model.fit(data, labels, epochs=5)
+        ```
+
+        You can pass this schedule directly into a
+        `tf.keras.optimizers.Optimizer` as the learning rate. The learning rate
+        schedule is also serializable and deserializable using
+        `tf.keras.optimizers.schedules.serialize` and
+        `tf.keras.optimizers.schedules.deserialize`.
```
Comment (seanpmorgan) [inline tensorflow_addons/optimizers/cyclical_learning_rate.py]: Hmmm these may be a bit redundant seeing as it's [written in the base class](https://github.com/tensorflow/tensorflow/blob/r2.0/tensorflow/python/keras/optimizer_v2/learning_rate_schedule.py#L114-L116) and there is already an example of using the LRScheduler above?

File: tensorflow_addons/optimizers/cyclical_learning_rate.py [hunk 2]
```diff
@@ -0,0 +1,305 @@
...
```
Comment (seanpmorgan) [inline tensorflow_addons/optimizers/cyclical_learning_rate.py]: ditto

File: tensorflow_addons/optimizers/cyclical_learning_rate.py [hunk 3]
```diff
@@ -0,0 +1,305 @@
...
```
Comment (seanpmorgan) [inline tensorflow_addons/optimizers/cyclical_learning_rate.py]: ditto

File: tensorflow_addons/optimizers/cyclical_learning_rate.py (hunk 1 above)
Comment (RaphaelMeudec) [reply tensorflow_addons/optimizers/cyclical_learning_rate.py]: Do you mean removing it from the docstring?

Comment (RaphaelMeudec) [reply tensorflow_addons/optimizers/cyclical_learning_rate.py]: I've removed the part which is common with `learning_rate_schedule.py`

File: tensorflow_addons/optimizers/README.md [hunk 4]
```diff
@@ -10,6 +10,7 @@
 | moving_average | Dheeraj R. Reddy | dheeraj98reddy@gmail.com |
 | rectified_adam | Zhao Hanguang | cyberzhg@gmail.com |
 | weight_decay_optimizers |  Phil Jund | ijund.phil@googlemail.com   |
+| cyclical_learning_rate | Raphael Meudec | raphael.meudec@gmail.com |
```
Comment (seanpmorgan) [inline tensorflow_addons/optimizers/README.md]: Alphabetize please

File: tensorflow_addons/optimizers/README.md [hunk 5]
```diff
@@ -22,6 +23,7 @@
 | moving_average | MovingAverage | |
 | rectified_adam | RectifiedAdam | https://arxiv.org/pdf/1908.03265v1.pdf |
 | weight_decay_optimizers | SGDW, AdamW, extend_with_decoupled_weight_decay | https://arxiv.org/pdf/1711.05101.pdf |
+| cyclical_learning_rate | Cyclical Learning Rate | https://arxiv.org/abs/1506.01186 |
```
Comment (seanpmorgan) [inline tensorflow_addons/optimizers/README.md]: Alphabetize please

File: tensorflow_addons/optimizers/README.md (hunk 4 above)
Comment (RaphaelMeudec) [reply tensorflow_addons/optimizers/README.md]: What do you mean by alphabetize?

Comment (seanpmorgan) [reply tensorflow_addons/optimizers/README.md]: Just order alphabetically:

Comment (seanpmorgan) [reply tensorflow_addons/optimizers/README.md]: i.e. place it below conditional gradient in the list

Comment (RaphaelMeudec) [reply tensorflow_addons/optimizers/README.md]: Done!

**Comment by RaphaelMeudec** [comment]
@seanpmorgan Alphabetized it and rebased on tensorflow/addons. You might want to re-run kokoro

**Comment by gabrieldemarmiesse** [comment]
Thanks a lot for the contribution @RaphaelMeudec  and thank you @seanpmorgan  for the review!
